lib/modules/datasource/python-version/index.ts

```diff
@@ -0,0 +1,98 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { id as versioning } from '../../versioning/python';
+import { Datasource } from '../datasource';
+import { EndoflifeDatePackagesource } from '../endoflife-date';
+import { registryUrl as eolRegistryUrl } from '../endoflife-date/common';
+import { GithubReleasesDatasource } from '../github-releases';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { datasource, defaultRegistryUrl, githubBaseUrl } from './common';
+import { PythonRelease } from './schema';
+
+export class PythonVersionDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+    PythonVersionDatasource.pythonPrebuildDatasource =
+      new GithubReleasesDatasource();
+    PythonVersionDatasource.pythonEolDatasource =
+      new EndoflifeDatePackagesource();
+  }
```

Comment:
make instance properties, otherwise they get overridden when a new instance of this class is created. or move them to common, so they instanciated once